src/app/services/userService.tsx

```diff
@@ -0,0 +1,27 @@
+export const fetchUserProfile = async (userId: string) => {
+    const response = await fetch(`/api/users/${userId}`, {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+    });
+    if (!response.ok) throw new Error('Error fetching user profile');
+    return response.json();
+};
+
+export const updateUserProfile = async (name: string, email: string, userId: string) => {
+    const response = await fetch(`/api/users/${userId}`, {
+        method: 'PATCH',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ name, email }),
+    });
+    if (!response.ok) throw new Error('Error updating profile');
+    return response.json();
+};
+
+export const fetchUsers = async() =>{
```

Comment:
Assuming this reutrns an array of users, the return type would be
```suggestion
export const fetchUsers = async(): Promise<Partial<User>[]> =>{
```